server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeRole.java

```diff
@@ -134,10 +134,21 @@ public final String toString() {
 
     };
 
+    public static final DiscoveryNodeRole REMOTE_CLUSTER_CLIENT_ROLE = new DiscoveryNodeRole("remote_cluster_client", "r") {
+
+        @Override
+        protected Setting<Boolean> roleSetting() {
+            return Node.NODE_REMOTE_CLUSTER_CLIENT;
+        }
+
+    };
+
     /**
      * The built-in node roles.
      */
-    public static Set<DiscoveryNodeRole> BUILT_IN_ROLES = Set.of(DATA_ROLE, INGEST_ROLE, MASTER_ROLE);
+    public static Set<DiscoveryNodeRole> BUILT_IN_ROLES = Set.of(DATA_ROLE, INGEST_ROLE, MASTER_ROLE, REMOTE_CLUSTER_CLIENT_ROLE);
+
+    static Set<DiscoveryNodeRole> LEGACY_ROLES = Set.of(DATA_ROLE, INGEST_ROLE, MASTER_ROLE);
```

Comment:
No, this shouldn't be based on version. We have two things:
 - very old nodes before we made roles pluggable that only understand three roles: master, data, and ingest
 - new nodes after we made roles pluggable that understand the roles they are built with us, but *also* understand that other nodes might send them roles that they don't understand
 - for example, a 7.6.0 node today while it doesn't have a role built-in for `remote_cluster_client`, *does* have the ability to receive that role from a 7.7.0 node, and represent it accordingly
 - however, a 7.0.0 node, before we made roles pluggable, has *no* possibility of understanding a `remote_cluster_client` role, it would blow up if it sees it

Reply:
Thanks for the explanation